libres/lib/include/ert/analysis/update.hpp

```diff
@@ -0,0 +1,102 @@
+#include <ert/enkf/local_dataset.hpp>
+#include <ert/enkf/enkf_config_node.hpp>
+#include <ert/enkf/ensemble_config.hpp>
+
+#include <ert/enkf/enkf_state.hpp>
+#include <ert/enkf/enkf_obs.hpp>
+
+/*
+Store a set of parameters into a enkf_fs_type storage
+*/
+void analysis_save_parameters(
+    enkf_fs_type *target_fs, ensemble_config_type *ensemble_config,
+    int_vector_type *iens_active_index, int last_step,
+    enkf_state_type **ensemble, hash_type *use_count,
+    const local_ministep_type *ministep,
+    std::unordered_map<std::string, matrix_type *> parameters);
+
+/*
+Store a set of row-scaled parameters into a enkf_fs_type storage
+*/
+void analysis_save_row_scaling_parameters(
+    enkf_fs_type *target_fs, ensemble_config_type *ensemble_config,
+    int_vector_type *iens_active_index, int last_step,
+    const local_ministep_type *ministep,
+    std::unordered_map<
+        std::string,
+        std::vector<std::pair<matrix_type *, const row_scaling_type *>>>
+        parameters);
+
+/*
+load a set of parameters from a enkf_fs_type storage into a set of
+matrices.
+*/
+std::unordered_map<std::string, matrix_type *> analysis_load_parameters(
+    enkf_fs_type *target_fs, ensemble_config_type *ensemble_config,
+    int_vector_type *iens_active_index, int last_step, meas_data_type *forecast,
+    enkf_state_type **ensemble, hash_type *use_count, obs_data_type *obs_data,
+    const local_ministep_type *ministep);
+
+/*
+load a set of parameters from a enkf_fs_type storage into a set of
+matrices with the corresponding row-scaling object.
+*/
+std::unordered_map<
+    std::string,
+    std::vector<std::pair<matrix_type *, const row_scaling_type *>>>
+analysis_load_row_scaling_parameters(
+    enkf_fs_type *target_fs, ensemble_config_type *ensemble_config,
+    int_vector_type *iens_active_index, int last_step, meas_data_type *forecast,
+    enkf_state_type **ensemble, hash_type *use_count, obs_data_type *obs_data,
+    const local_ministep_type *ministep);
+
+/*
+Run the row-scaling enabled update algorithm on a set of A matrices.
+*/
+void analysis_run_analysis_update_with_rowscaling(
+    analysis_module_type *module, const bool_vector_type *ens_mask,
+    const meas_data_type *forecast, obs_data_type *obs_data,
+    rng_type *shared_rng, matrix_type *E,
+    std::unordered_map<
+        std::string,
+        std::vector<std::pair<matrix_type *, const row_scaling_type *>>>
+        parameters);
+
+/*
+Run the update algorithm on a set of A matrices without row-scaling
+*/
+void analysis_run_analysis_update(
+    analysis_module_type *module, const bool_vector_type *ens_mask,
+    const meas_data_type *forecast, obs_data_type *obs_data,
+    rng_type *shared_rng, matrix_type *E,
+    std::unordered_map<std::string, matrix_type *> parameters);
+
+/*
+Check whether the current state and config allows the update algorithm
+to be executed
+*/
+bool analysis_assert_update_viable(const analysis_config_type *analysis_config,
```

Comment:
This function seems to be somewhat unlike the other ones as it was just a static function in `enkf_main.cpp`. Why is it being moved?